sql/core/src/main/scala/org/apache/spark/sql/execution/pythonUDFs.scala

```diff
@@ -220,6 +214,89 @@ object EvaluatePython {
     // TODO(davies): we could improve this by try to cast the object to expected type
     case (c, _) => null
   }
+
+
+  private val module = "pyspark.sql.types"
+
+  /**
+   * Pickler for StructType
+   */
+  private class StructTypePickler extends IObjectPickler {
+
+    private val cls = classOf[StructType]
+
+    def register(): Unit = {
+      Pickler.registerCustomPickler(cls, this)
+    }
+
+    def pickle(obj: Object, out: OutputStream, pickler: Pickler): Unit = {
+      out.write(Opcodes.GLOBAL)
+      out.write((module + "\n" + "_parse_datatype_json_string" + "\n").getBytes)
```

Comment:
Should we pass an explicit character set to `getBytes`?  I worry that falling back to the platforms default encoding might cause problems if the platform uses a wider encoding than Python expects.